File: examples/micronaut-app/src/test/java/ht/eyfout/example/http/ExampleHttpAPI.java
package ht.eyfout.example.http;

import ht.eyfout.junit.jupiter.api.http.HttpAPI;
import ht.eyfout.junit.jupiter.api.http.HttpAPIRequestExecutor;

public class ExampleHttpAPI extends HttpAPI<HttpAPIRequestExecutor> {
    public static HttpAPI<HttpAPIRequestExecutor> INSTANCE = new ExampleHttpAPI();

    public ExampleHttpAPI() {

    }

    @Override
    public String getHttpMethod() {
        return "GET";
    }

    @Override
    public String getBasePath() {
        return "v1/vehicles";
    }

    @Override
    public String getDescription() {
        return "query";
    }
}

File: examples/micronaut-app/src/test/kotlin/ht/eyfout/example/http/ExampleHttpAPI.kt
package ht.eyfout.example.http

import ht.eyfout.junit.jupiter.api.http.HttpAPI
import ht.eyfout.junit.jupiter.api.http.HttpAPIRequestExecutor

class ExampleHttpAPI : HttpAPI<HttpAPIRequestExecutor>() {
    companion object {
        val INSTANCE = ExampleHttpAPI()
    }

    override fun getHttpMethod(): String = "GET"

    override fun getBasePath(): String = "v1/vehicles"

    override fun getDescription(): String = "vehicles"
}